libraries/lib-wx-init/SettingsWX.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "BasicSettings.h"
+
+#include <memory>
+#include <wx/string.h>
+#include <wx/arrstr.h>
+
+class wxConfigBase;
+
+class WX_INIT_API SettingsWX final : public audacity::BasicSettings
+{
+   wxArrayString mGroupStack;
+   std::unique_ptr<wxConfigBase> mConfig;
+   const bool mIsConfigOwner;
+protected:
+   void DoBeginGroup(const wxString& prefix) override;
+   void DoEndGroup() noexcept override;
+
+public:
+   ///@brief Constructs default application settings
+   SettingsWX();
+   SettingsWX(std::unique_ptr<wxConfigBase> config);
```

Comment:
`unique_ptr` makes a lot of sense here, as you express that SettingsWX takes the config ownership from the caller. However, I think `mConfig` should be a shared pointer. Luckily, shared_ptr has a constructor that consumes `unique_ptr`.

Also, I vote for marking the constructor `explicit`.

Reply:
Not sure if I understand that, but I attempted to change ownership policy in a new separate commit